src/components/SpatialBookMarks.vue

```diff
@@ -0,0 +1,191 @@
+<!--
+  @file
+  @since v3.8
+-->
+
+<template>
+  <ul
+    id="g3w-spatial-bookmarks"
+    class="treeview-menu g3w-spatial-bookmarks menu-items"
+    :class="{'g3w-tools': !showaddform}"
+  >
+    <template v-if="showaddform">
+
+      <li>
+        <div style="display: flex; justify-content: end">
+
+          <span
+            v-t-tooltip:left.create="'close'"
+            @click.stop="showaddform = false"
+            :class="g3wtemplate.getFontClass('close')"
+            class="sidebar-button sidebar-button-icon"
+            style="padding: 5px; margin: 3px;"
+          >
+          </span>
+
+        </div>
+
+        <helpdiv message="sdk.spatialbookmarks.helptext"/>
+        <div
+          class="container add-bookmark-input"
+          style="padding: 5px; width: 100%"
+        >
+          <input-text
+            :state="addbookmarkinput"/>
```

Comment:
here I thought of adding something like `@keyup.enter="addBookMark"`, to be able to add a bookmark even just by pressing the enter key.

Do you think it's somehow feasible or it requires too many modifications elsewhere?

Reply:
@Raruto for the moment leave it as it is. Need to click on button to add bookmark